frontend/ui/widget/filechooser.lua

```diff
@@ -56,12 +56,14 @@ function FileChooser:init()
         end
         return true
     end
-    self.list = function(path, dirs, files)
+    self.list = function(path, dirs, files, count_only)
         -- lfs.dir directory without permission will give error
         local ok, iter, dir_obj = pcall(lfs.dir, path)
         if ok then
             for f in iter, dir_obj do
-                if self.show_hidden or not string.match(f, "^%.[^.]") then
+                if count_only then
+                    table.insert(dirs, true)
+                elseif self.show_hidden or not string.match(f, "^%.[^.]") then
```

Comment:
If `count_only`, shouldn't we count only the not hidden ones if `not show_hidden`?
So the user don't see `Directory name ..... 13 items` and when entering the directory, possibly see only 4 files.

Reply:
At a quick read, I'll fear we might take the elseif branch even if count_only.
Shouldn't it be:
```lua
if count_only then
  if ... check whether hidden ... then
    table.insert(dirs, true)
  end
elseif .. the original checks ... then
  .. the original stuff ...
end
```

Reply:
If I'm not mistaken the Windows File Manager (or perhaps the one that came with DOS, or an early version of Explorer) did something like that. I always liked it as a kind of insider secret. ;-)

Reply:
(unresolved that conversation, as our last comments might have been simultaneous?)

Reply:
You're right, that wasn't an issue the way I wrote it not caring about that, but of course it is now.